processor/src/test/java/org/mapstruct/ap/test/hascheck/HasCheckTest.java

```diff
@@ -0,0 +1,113 @@
+/**
+ *  Copyright 2012-2015 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.hascheck;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner;
+
+/**
+ * Test for correct handling of has checks.
+ *
+ * @author Sean Huang
+ */
+@WithClasses({
+    SourceTargetMapper.class,
+    SourceTargetWtCheckMapper.class,
+    MyObject.class,
+    CustomMapper.class,
+    MyLongWrapper.class,
+    Source.class,
+    Target.class,
+    SourceWtCheck.class,
+    TargetWtCheck.class
+})
+@RunWith(AnnotationProcessorTestRunner.class)
+public class HasCheckTest {
+    @Test //Can't test seeing exception if config is on, as there is compilation error
```

Comment:
What compilation error is this? Generally we should prevent to generate non-compiling code (if that's what's happening) and rather raise a meaningful error from within the generator.